component/src/main/java/io/siddhi/extension/io/file/listeners/FileCronListener.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.siddhi.extension.io.file.listeners;
+
+import io.siddhi.core.stream.input.source.SourceEventListener;
+import io.siddhi.extension.io.file.processors.FileProcessor;
+import io.siddhi.extension.io.file.util.Constants;
+import io.siddhi.extension.io.file.util.FileSourceConfiguration;
+import io.siddhi.extension.io.file.util.VFSClientConnectorCallback;
+import org.apache.log4j.Logger;
+import org.quartz.CronScheduleBuilder;
+import org.quartz.Job;
+import org.quartz.JobBuilder;
+import org.quartz.JobDataMap;
+import org.quartz.JobDetail;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.JobKey;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.Trigger;
+import org.quartz.TriggerBuilder;
+import org.quartz.impl.StdSchedulerFactory;
+import org.wso2.carbon.messaging.exceptions.ClientConnectorException;
+import org.wso2.transport.file.connector.sender.VFSClientConnector;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * FileCron Listener is executted when the cron expression is given. If the current time satisfied by the cron
+ * expression then the file processing will be executed.
+ */
+public class FileCronListener implements Job {
```

Comment:
Shall we rename this class as FileCronReader? since this is not an actual listener class?

Reply:
fixed in 47cc199b8188f3701d228a031aeb7a33f02bc856